app/src/main/java/io/appium/uiautomator2/utils/ElementHelpers.java

```diff
@@ -84,8 +87,41 @@ private static AccessibilityNodeInfo elementToNode(Object element) {
      *
      * For example, appium returns elements like [{"ELEMENT":1}, {"ELEMENT":2}]
      */
-    public static JSONObject toJSON(AndroidElement el) throws JSONException {
-        return new JSONObject().put("ELEMENT", el.getId());
+    public static JSONObject toJSON(AndroidElement el) throws JSONException, UiObjectNotFoundException {
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("ELEMENT", el.getId());
+        if (!Session.shouldUseCompactResponses()) {
+            for (String field : Session.getElementResponseFields()) {
+                try {
+                    if (field.equals("name")) {
+                        putNullable(jsonObject, field, el.getContentDesc());
+                    } else if (field.equals("text")) {
+                        putNullable(jsonObject, field, el.getText());
+                    } else if (field.equals("rect")) {
+                        putNullable(jsonObject, field, GetRect.getElementRectJSON(el));
+                    } else if (field.equals("enabled")) {
+                        putNullable(jsonObject, field, GetElementAttribute.getElementAttributeValue(el, field));
+                    } else if (field.equals("displayed")) {
+                        putNullable(jsonObject, field, GetElementAttribute.getElementAttributeValue(el, field));
+                    } else if (field.equals("selected")) {
+                        putNullable(jsonObject, field, GetElementAttribute.getElementAttributeValue(el, field));
+                    } else if (field.startsWith("attribute/")) {
```

Comment:
please move the string into a constant

Reply:
All strings, or just "attribute/" ?

Reply:
I'm concerned only about this one, since other strings are only used once in the code, but for this one you need to calculate the length later